addon/mixins/route.js

```diff
@@ -216,42 +124,49 @@ const RouteMixin = Mixin.create({
       delete options.storeFindMethod;
     }
 
-    const startingPage = options.startingPage === undefined ? 0 : options.startingPage-1;
-    const perPage      = options.perPage || this.get('_perPage');
-    const modelPath    = options.modelPath || this.get('_modelPath');
+    const currentPage = options.startingPage === undefined ? 0 : options.startingPage-1;
+    const perPage = options.perPage || 25;
+
+    // check if user passed in param w/ infinityModel, else check if defined on the route (for backwards compat), else default
+    const perPageParam = typeOfCheck(options.perPageParam, get(this, 'perPageParam'), 'per_page');
+    const pageParam = typeOfCheck(options.pageParam, get(this, 'pageParam'), 'page');
+    const totalPagesParam = options.totalPagesParam || 'meta.total_pages';
```

Comment:
Can we do an Ember#deprecate in the case that one of the properties was defined on the route - bugging the developer to update to the new format?

https://www.emberjs.com/api/ember/2.14/namespaces/Ember/methods/deprecate?anchor=deprecate

Probably makes most sense to do this in the util rather than here.